### PR TITLE
Add Uniformat Table

### PR DESCRIPTION
**Pull request by axelstudios**
#### What's this PR do?
Adds all uniformat metadata to a dedicated table for other data to reference

#### How should this be manually tested?
1. Run migrations
2. Inspect the new `seed_uniformat` table

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: requirements/aws.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@
 boto3==1.17.36
 django-sslify==0.2.7
 django-ses==3.5.0
-uWSGI==2.0.22
+uWSGI==2.0.22; sys_platform != "win32"
```
Comment (nllong) [inline requirements/aws.txt:5]: so does windows not need this to run?

File: seed/lib/uniformat/uniformat.py [hunk 2]
```diff
@@ -0,0 +1,4418 @@
...
+SEED Platform (TM), Copyright (c) Alliance for Sustainable Energy, LLC, and other contributors.
+See also https://github.com/seed-platform/seed/main/LICENSE.md
+"""
+from django.db import transaction
+from typing_extensions import NotRequired, TypedDict
+
+from seed.models import Uniformat
+
+
+class UniformatEntry(TypedDict):
```
Comment (nllong) [inline seed/lib/uniformat/uniformat.py:13]: fancy typing, cool

File: seed/models/uniformat.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class Uniformat(models.Model):
+    code = models.CharField(max_length=7, unique=True)
+    category = models.CharField(max_length=100)
+    definition = models.CharField(max_length=1024, null=True)
+    imperial_units = models.CharField(max_length=10, null=True)
+    metric_units = models.CharField(max_length=10, null=True)
+    quantity_definition = models.CharField(max_length=100, null=True)
+    parent = models.ForeignKey('self', null=True, on_delete=models.CASCADE)
```
Comment (nllong) [inline seed/models/uniformat.py:17]: nice, linked list

File: seed/lib/uniformat/uniformat.py [hunk 4]
```diff
@@ -0,0 +1,4418 @@
...
+    metric_units: NotRequired[str]
+    quantity_definition: NotRequired[str]
+
+
+uniformat_data: list[UniformatEntry] = [
+    {
+        'code': 'A',
+        'category': 'SUBSTRUCTURE',
+        'definition': 'This system includes all work below the lowest floor construction (usually slab-on-grade) and the enclosing horizontal and vertical elements required to form a basement, together with the necessary mass excavation and backfill.',
+        'imperial_units': 'SF',
```
Comment (nllong) [inline seed/lib/uniformat/uniformat.py:27]: this is fine for now, but we should use the format that is defined in pint (e.g., ft**2)

**Review by nllong — APPROVED** [review]

File: requirements/aws.txt (hunk 1 above)
Comment (axelstudios) [reply requirements/aws.txt:5]: Not only does it not need it, they don't _have_ a Windows version, so it always fails with `pip install`

File: seed/lib/uniformat/uniformat.py (hunk 4 above)
Comment (axelstudios) [reply seed/lib/uniformat/uniformat.py:27]: I'm happy to change it, everything is verbatim from https://www.wbdg.org/ffc/navy-navfac/design-build-request-proposal/uniformat-structure

**Pull request merged**